### PR TITLE
Add CRUD for topics

### PR DESCRIPTION
**Pull request by Maru321**
#### Description:
- Added topics' CRUD for admins to be able to create them

---
#### Tasks:
- [x] Add the topic model with its migrations, add the active admin topic.rb file and modify the user.rb active admin file

---
#### Risk:
- Low

---
#### Preview:
- N/A

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -16,4 +16,6 @@
       resources :users, only: [:update]
     end
   end
+
+  resources :topics
```
Comment (Maru321) [inline config/routes.rb]: It is not under the V1 namespace because it is a shared resource with admin views

File: Gemfile [hunk 2]
```diff
@@ -25,7 +25,7 @@ gem 'pry-rails', '~> 0.3.4'
 gem 'rails_12factor', '~> 0.0.3', group: :production
 gem 'rack-cors', '~> 0.4.0', require: 'rack/cors'
 gem 'simple_token_authentication', '~> 1.6.0'
-
+gem 'paranoia'
```
Comment (MaicolBen) [inline Gemfile]: add version

File: app/admin/topic.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
+ActiveAdmin.register Topic do
+  # See permitted parameters documentation:
+  # https://github.com/activeadmin/activeadmin/blob/master/docs/2-resource-customization.md#setting-up-strong-parameters
+  #
+  permit_params :name
+  #
+  # or
+  #
+  # permit_params do
```
Comment (MaicolBen) [inline app/admin/topic.rb]: Remove unused code

File: Gemfile (hunk 2 above)
Comment (matiasmansilla1989) [inline Gemfile]: Add a brief explanation about why are you using `paranoia` would be great.

File: app/models/topic.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# == Schema Information
+#
+# Table name: topics
+#
+#  id         :integer          not null, primary key
+#  name       :string
```
Comment (matiasmansilla1989) [inline app/models/topic.rb]: Are you going to add validations in a next PR? I think that the name should be mandatory and unique

File: Gemfile (hunk 2 above)
Comment (MaicolBen) [inline Gemfile]: You can read the github readme, or add [label](https://github.com/jgorset/label) to your project

File: db/migrate/20160805171111_create_topics.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class CreateTopics < ActiveRecord::Migration
+  def change
+    create_table :topics do |t|
+      t.string :name
```
Comment (matiasmansilla1989) [inline db/migrate/20160805171111_create_topics.rb]: when adding presence validations add a NOT NULL constraint here in order to be consistent

File: spec/factories/topics.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+# Indexes
+#
+#  index_topics_on_deleted_at  (deleted_at)
+#
+
+FactoryGirl.define do
+  factory :topic do
+    name "MyString"
+  end
+
```
Comment (matiasmansilla1989) [inline spec/factories/topics.rb]: remove this extra space

File: Gemfile (hunk 2 above)
Comment (matiasmansilla1989) [inline Gemfile]: @MaicolBen I don't understand your comment. I mean Why `paranoia` is necessary?
(I know the answer but it's just for adding some context to other people)

Comment (MaicolBen) [inline Gemfile]: Oh I thought that you wanted to know what the gem does, but if you want why is she adding it, I agree with you that there is needed an explanation for it. Maybe you don't need a gem for it.

File: spec/factories/topics.rb [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+#  deleted_at :datetime
+#
+# Indexes
+#
+#  index_topics_on_deleted_at  (deleted_at)
+#
+
+FactoryGirl.define do
+  factory :topic do
+    name "MyString"
```
Comment (adicaff) [inline spec/factories/topics.rb]: Single quotes and use Faker if you can too.

File: Gemfile (hunk 2 above)
Comment (matiasmansilla1989) [inline Gemfile]: We have not discussed it. I agree with you that, most of the things that `paranoia` does, can be done manually, but this is just a training project, it's not so bad using `paranoia`

**Comment by MaicolBen** [comment]
It looks good to me!

**Pull request merged**